### PR TITLE
feat: multiplication

### PR DESCRIPTION
**Pull request by henryiii**
Also adds some typing, and various fixes / updates.

### CODE REVIEW

File: setup.cfg [hunk 1]
```diff
@@ -61,6 +62,12 @@ warn_unused_ignores = True
 [mypy-numpy]
 ignore_missing_imports = True
 
+[mypy-numba]
+ignore_missing_imports = True
+
+[mypy-awkward1]
```
Comment (henryiii) [inline setup.cfg:68]:
```suggestion
[mypy-awkward]
```

File: src/vector/core/lorentz/all.py [hunk 2]
```diff
@@ -4,34 +4,59 @@
...
 
 def rapidity(vec):
+    # type: (LorentzVector) -> Scalar
     return 0.5 * np.log((vec.t + vec.z) / (vec.t - vec.z))
 
 
-def delta_r(vec, other):
+def delta_r2(vec, other):
+    # type: (LorentzVector, LorentzVector) -> Scalar
     """Return :math:`\\Delta R` the distance in (eta,phi) space with another Lorentz vector, defined as:
```
Comment (eduardo-rodrigues) [inline src/vector/core/lorentz/all.py:50]:
```suggestion
    """Return :math:`\\Delta R^2` the distance squared in (eta,phi) space with another Lorentz vector, defined as:
```

File: src/vector/core/lorentz/all.py [hunk 3]
```diff
@@ -4,34 +4,59 @@
...
     :math:`\\Delta R = \\sqrt{(\\Delta \\eta)^2 + (\\Delta \\phi)^2}`
```
Comment (eduardo-rodrigues) [inline src/vector/core/lorentz/all.py:51]:
```suggestion
    :math:`\\Delta R^2 = (\\Delta \\eta)^2 + (\\Delta \\phi)^2`
```

Comment (henryiii) [reply src/vector/core/lorentz/all.py:51]: 😳

**Pull request merged**